### PR TITLE
Parentheses in macro args prevent commas from starting new arguments

### LINKED ISSUE #704: [Feature request] Macro arguments should allow commas inside parentheses
Forking off a point @aaaaaa123456789 brought up in the #685 comments.

Commas inside parentheses are most likely to occur when separating function arguments, e.g. `ATAN2(1.0, 2.0)` or `STRSUB("abc", 3, 1)`. It would be convenient if lexing macro arguments kept track of the parentheses depth, and did not have to `\,` escape commas when inside parentheses. To explicitly prevent this, one could `\(` escape an opening parenthesis instead.

This would also bring macro behavior further in line with built-in commands, so that `println "a,b", atan2(1,2)` and `myprintmac "a,b", atan2(1,2)` would behave the same way.

For example:

```
mac: MACRO
    println "\1 = ", \1
    println "\2 and \3"
ENDM
    mac mul(3.0, 4.0) >> 16, sad :\,\(, happy :)
```
```
mul(3.0, 4.0) >> 16 = $C
sad :,( and happy :)
```

This would be backwards-compatible with the common case: `mac atan2(1\,2)` would continue to work, and now so would `mac atan2(1,2)`. It would be incompatible with (what I expect is) the uncommon case: `mac (a, b)` would pass one arg instead of two, and would need editing to `mac \(a, b)`.

I'd implement this after #685, so parentheses inside string literals would not be counted.

### PR DESCRIPTION
**Pull request by Rangi42**
This is similar to C's behavior, and convenient for passing function calls as single values, like `MUL(3.0, 4.0)` or `STRSUB("str", 2, 1)`.

Fixes #704

### CODE REVIEW

File: src/asm/lexer.c [hunk 1]
```diff
@@ -2039,6 +2039,7 @@ static int yylex_RAW(void)
 		 lexer_GetLineNo(), lexer_GetColNo());
 
 	/* This is essentially a modified `appendStringLiteral` */
+	unsigned int parenDepth = 0;
```
Comment (ISSOtm) [inline src/asm/lexer.c:2042]: Why `unsigned int`, and not e.g. `size_t`?

File: test/asm/raw-macro-args.asm [hunk 2]
```diff
@@ -22,7 +22,7 @@ STR EQUS "str\"ing"
 	printargs "literal \"\\\"", \ ; comment 2
 """multi-"line"
   ""string"" arg"""
-	printargs MUL(2.0\, 3.0)
+	printargs MUL(2.0, 3.0)
```
Comment (ISSOtm) [inline test/asm/raw-macro-args.asm:25]: Shouldn't both unescaped and escaped work?

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/asm/lexer.c (hunk 1 above)
Comment (Rangi42) [reply src/asm/lexer.c:2042]: No particular reason. Should I change it?

File: test/asm/raw-macro-args.asm (hunk 2 above)
Comment (Rangi42) [reply test/asm/raw-macro-args.asm:25]: Yes, I just removed this comma because it was no longer necessary.

File: src/asm/lexer.c (hunk 1 above)
Comment (ISSOtm) [reply src/asm/lexer.c:2042]: Please do.

File: test/asm/raw-macro-args.asm (hunk 2 above)
Comment (ISSOtm) [reply test/asm/raw-macro-args.asm:25]: Yes, but then I think both should be tested, for backwards compatibility's sake.

Comment (Rangi42) [reply test/asm/raw-macro-args.asm:25]: I've added one in `a(b,c\,d)`.

File: src/asm/rgbasm.5 [hunk 3]
```diff
@@ -1184,7 +1184,11 @@ ENDM
 .Pp
 Macro arguments support all the escape sequences of strings, as well as
 .Ql \[rs],
-to escape commas, since those otherwise separate arguments.
+to escape commas, and
```
Comment (ISSOtm) [inline src/asm/rgbasm.5:1187]:
```suggestion
to escape commas, as well as
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**